### PR TITLE
Homework - bus ticket

### PR DESCRIPTION
**Pull request by gujaKhela**

### CODE REVIEW

File: src/BusTicket.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+        this.ticketClass = ticketClass;
+        this.ticketType = ticketType;
+        this.startDate = startDate;
+        this.price = (price != null) ? price : 0.0;
+    }
+
+    public String getTicketClass() { return ticketClass; }
+    public String getTicketType() { return ticketType; }
+    public LocalDate getStartDate() { return startDate; }
+    public double getPrice() { return price; }
```
Comment (Alexmansar) [inline src/BusTicket.java]:
```
public String getTicketClass() {
     return ticketClass; 
}
```

update the formattingg in each method

File: src/TicketReader.java [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+                                    startDateStr = !value.equals("null") ? value : null;
+                                    break;
+                                case "price":
+                                    try {
+                                        price = !value.equals("null") && !value.isEmpty() ? Double.parseDouble(value) : null;
+                                    } catch (NumberFormatException e) {
+                                        System.out.println("Invalid price format: " + value);
+                                    }
+                                    break;
+                            }
```
Comment (Alexmansar) [inline src/TicketReader.java]: why java version you use?

File: src/TicketValidator.java [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+    public boolean validateTicket(BusTicket ticket) {
+        boolean isValid = true;
+
+        // Validate startDate
+        if (ticket.getStartDate() != null && ticket.getStartDate().isAfter(LocalDate.now())) {
+            System.out.println("Error: Start date can't be in the future.");
+            startDateViolationCount++;
+            isValid = false;
+        }
```
Comment (Alexmansar) [inline src/TicketValidator.java]: create separate methods for each validation please

File: src/TicketValidator.java [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+    public String getMostPopularViolation() {
+        if (startDateViolationCount >= priceViolationCount && startDateViolationCount >= ticketTypeViolationCount) {
+            return "start date";
+        } else if (priceViolationCount >= ticketTypeViolationCount) {
+            return "price";
+        } else {
+            return "ticket type";
+        }
+    }
```
Comment (Alexmansar) [inline src/TicketValidator.java:37]: move it before private methods

**Review by Alexmansar — CHANGES_REQUESTED** [review]
@gujaKhela Hi, please check comments)) In general good job

File: src/TicketReader.java (hunk 2 above)
Comment (gujaKhela) [reply src/TicketReader.java]: i am using Java 21, should i used old versions?

Comment (Alexmansar) [reply src/TicketReader.java]: @gujaKhela  Its ok)) but you use old style)) https://www.baeldung.com/java-switch#1-the-new-switch-expression
try use new one please

**Review by Alexmansar — APPROVED** [review]
@gujaKhela Could be merged)

**Pull request merged**